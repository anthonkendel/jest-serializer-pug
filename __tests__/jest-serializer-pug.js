const jestSerializerPug = require('../jest-serializer-pug');
const html = require('../html');

expect.addSnapshotSerializer(jestSerializerPug);
test('type: string', () => {
  expect(html.string).toMatchSnapshot();
});

test('type: String', () => {
  expect(html.stringClass).toMatchSnapshot();
});

test('type: Element', () => {
  expect(html.element).toMatchSnapshot();
});
