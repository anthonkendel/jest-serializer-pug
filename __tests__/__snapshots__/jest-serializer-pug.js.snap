// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type: Element 1`] = `
div
  .field
    label.label Username
    .control.has-icons-left.has-icons-right
      input.input.is-success(type="text", placeholder="Text input", value="bulma")
      span.icon.is-small.is-left
        i.fas.fa-user
      span.icon.is-small.is-right
        i.fas.fa-check
    p.help.is-success This username is available
  .field
    label.label Message
    .control
      textarea.textarea(placeholder="Textarea")
  .field
    .control
      label.checkbox
        input(type="checkbox")
        | 
I agree to the 
        a(href="#") terms and conditions
  .field
    .control
      label.radio
        input(type="radio", name="question")
        | 
Yes

      label.radio
        input(type="radio", name="question")
        | 
No

  .field.is-grouped
    .control
      button.button.is-link Submit
    .control
      button.button.is-link.is-light Cancel
`;

exports[`type: String 1`] = `
div
  .field
    label.label Username
    .control.has-icons-left.has-icons-right
      input.input.is-success(type="text", placeholder="Text input", value="bulma")
      span.icon.is-small.is-left
        i.fas.fa-user
      span.icon.is-small.is-right
        i.fas.fa-check
    p.help.is-success This username is available
  .field
    label.label Message
    .control
      textarea.textarea(placeholder="Textarea")
  .field
    .control
      label.checkbox
        input(type="checkbox")
        | 
I agree to the 
        a(href="#") terms and conditions
  .field
    .control
      label.radio
        input(type="radio", name="question")
        | 
Yes

      label.radio
        input(type="radio", name="question")
        | 
No

  .field.is-grouped
    .control
      button.button.is-link Submit
    .control
      button.button.is-link.is-light Cancel
`;

exports[`type: string 1`] = `
div
  .field
    label.label Username
    .control.has-icons-left.has-icons-right
      input.input.is-success(type="text", placeholder="Text input", value="bulma")
      span.icon.is-small.is-left
        i.fas.fa-user
      span.icon.is-small.is-right
        i.fas.fa-check
    p.help.is-success This username is available
  .field
    label.label Message
    .control
      textarea.textarea(placeholder="Textarea")
  .field
    .control
      label.checkbox
        input(type="checkbox")
        | 
I agree to the 
        a(href="#") terms and conditions
  .field
    .control
      label.radio
        input(type="radio", name="question")
        | 
Yes

      label.radio
        input(type="radio", name="question")
        | 
No

  .field.is-grouped
    .control
      button.button.is-link Submit
    .control
      button.button.is-link.is-light Cancel
`;
